### DISCUSSION TITLE
Date Range component with ranges in the future for tracking incoming events (ex: next 7 days, next month, ...)

### DISCUSSION THREAD

**Discussion by lolcabanon** [Budibase/budibase#10316] [Ideas]
Hey! I was wondering why the native `Date Range` component does not offer to filter by **future dates**. I can't be the first to need this functionality, right?

Like for example, if I add an entry today (D day) and I already know I will have to do a follow-up in 30 days (D+30). Wouldn't it be nice to add this feature to the Date Range component, so it is actually possible to show a summary of the events due in the near future?

I guess it is possible following [this technique](https://github.com/Budibase/budibase/discussions/2859) to filter the due date by _start date_ and _end date_, but it's not really convenient (or user-friendly, especially to not-so-tech-savvy users). Since the Date Range component already exist, I guess it would be possible to add preconfigured ranges like "next 7 days", "next 30 days", and so on...

Please let me know!

Thanks :)

**Comment by melohagan** [comment]
Makes sense! The Date Range component itself isn't great being just a fixed dropdown. It would be nicer if you could select a range - which Flatpickr (the date library we use) supports.

I've raised an issue here: https://github.com/Budibase/budibase/issues/10334
